youtube_dl/extractor/moviefap.py

```diff
@@ -0,0 +1,123 @@
+        width = str_to_int(timeline.find('imageWidth').text)
+        height = str_to_int(timeline.find('imageHeight').text)
+        first = str_to_int(timeline.find('imageFirst').text)
+        last = str_to_int(timeline.find('imageLast').text)
+        pattern = timeline.find('imagePattern').text
+
+        # generate the list of thumbnail information dicts
+        thumbnails = []
+        for i in range(first, last + 1):
+            thumbnails.append({
+                'url': pattern.replace('#', str(i)),
+                'width': width,
+                'height': height
+            })
+        return thumbnails
+
+    def _real_extract(self, url):
+
+        # find the video ID
+        video_id = self._match_id(url)
+
+        # retrieve the page HTML
+        webpage = self._download_webpage(url, video_id)
+
+        # find the URL of the XML document detailing video download URLs
+        info_url = self._html_search_regex(r'flashvars\.config = escape\("(.+?)"', webpage, 'player parameters')
+
+        # download that XML
+        xml = self._download_xml(info_url, video_id)
+
+        # create dictionary of properties we know so far, or can find easily
+        info = {
+            'id': video_id,
+            'title': self._html_search_regex(r'<div id="view_title"><h1>(.*?)</h1>', webpage, 'title'),
+            'display_id': re.compile(self._VALID_URL).match(url).group('name'),
+            'thumbnails': self.__get_thumbnail_data(xml),
+            'thumbnail': xml.find('startThumb').text,
+            'description': self._html_search_regex(r'name="description" value="(.*?)"', webpage, 'description'),
+            'uploader_id': self._html_search_regex(r'name="username" value="(.*?)"', webpage, 'uploader_id'),
+            'view_count': str_to_int(self._html_search_regex(r'<br>Views <strong>([0-9]+)</strong>', webpage, 'view_count')),
+            'average_rating': float(self._html_search_regex(r'Current Rating<br> <strong>(.*?)</strong>', webpage, 'average_rating')),
+            'comment_count': str_to_int(self._html_search_regex(r'<span id="comCount">([0-9]+)</span>', webpage, 'comment_count')),
+            'age_limit': 18,
+            'webpage_url': self._html_search_regex(r'name="link" value="(.*?)"', webpage, 'webpage_url'),
+            'categories': self._html_search_regex(r'</div>\s*(.*?)\s*<br>', webpage, 'categories').split(', ')
+        }
+
+        # find and add the format
+        if xml.find('videoConfig') is not None:
+            info['ext'] = xml.find('videoConfig').find('type').text
+        else:
+            info['ext'] = 'flv'  # guess...
+
+        # work out the video URL(s)
+        if xml.find('videoLink') is not None:
+            # single format available
+            info['url'] = xml.find('videoLink').text
+        else:
+            # multiple formats available
+            info['formats'] = []
+
+            # N.B. formats are already in ascending order of quality
```

Comment:
The functionality of `_sort_formats` is very ambiguous. Are you saying I should call `self._sort_formats(info['formats'])` before `return info`? If so, would it be better for youtube-dl to do this automatically rather than delegate responsibility to each extractor?